File: 프로그래머스/1/161990. 바탕화면 정리/바탕화면 정리.js
function solution(wallpaper) {
    const R = wallpaper.length;
    const C = wallpaper[0].length;

    const rows = Array(R).fill(0);
    const cols = Array(C).fill(0);

    for (let r = 0; r < R; r++) {
        for (let c = 0; c < C; c++) {
            if (wallpaper[r][c] === '#') {
                rows[r]++;
                cols[c]++;
            }
        }
    }

    const strR = rows.findIndex(n => n > 0);
    const strC = cols.findIndex(n => n > 0);

    let endR = -1;
    for (let i = R - 1; i >= 0; i--) {
        if (rows[i] > 0) {
            endR = i;
            break;
        }
    }

    let endC = -1;
    for (let i = C - 1; i >= 0; i--) {
        if (cols[i] > 0) {
            endC = i;
            break;
        }
    }

    return [strR, strC, endR + 1, endC + 1];
}

File: 프로그래머스/1/161990. 바탕화면 정리/바탕화면 정리.py
def solution(wallpaper):
    R, C = len(wallpaper), len(wallpaper[0])
    
    rows, cols = [0] * R, [0] * C
    
    for r in range(R):
        for c in range(C):
            if wallpaper[r][c] == '#':
                rows[r] += 1
    
    for c in range(C):
        for r in range(R):
            if wallpaper[r][c] == '#':
                cols[c] += 1
            
    rIdxs = [i for i, n in enumerate(rows) if n > 0]
    cIdxs = [i for i, n in enumerate(cols) if n > 0]
            
    return [rIdxs[0], cIdxs[0], rIdxs[-1] + 1, cIdxs[-1] + 1]